apps/src/templates/sectionsRefresh/CoteacherSettings.jsx

```diff
@@ -33,36 +74,152 @@ export default function CoteacherSettings({
     setInputValue('');
   };
 
+  const handleInputChange = event => {
+    setInputValue(event.target.value);
+  };
+
   const handleSubmitAddEmail = e => {
     if (e.key === 'Enter') {
       e.preventDefault();
       handleAddEmail(e);
     }
   };
 
-  // coteacher count is teachers to add + the existing teachers - the primary teacher.
-  // If sectionInstructors is populated, we assume one of them is the primary teacher.
-  const existingCoteachers = sectionInstructors
-    ? sectionInstructors.length - 1
-    : 0;
-  const numCoteachers = existingCoteachers + coteachersToAdd.length;
+  const pill = (text, className, icon) => {
+    return (
+      <div className={classNames(className, styles.tablePill)}>
+        <StrongText>
+          <FontAwesome icon={icon} className={styles.tablePillIcon} />
+          {text}
+        </StrongText>
+      </div>
+    );
+  };
+
+  const statusPill = status => {
+    if (!status || status === 'invited') {
+      return pill(i18n.coteacherPending(), styles.tablePending, 'ellipsis');
+    } else if (status === 'active') {
+      return pill(i18n.coteacherAccepted(), styles.tableActive, 'check');
+    } else if (status === 'declined') {
+      return pill(i18n.coteacherDeclined(), styles.tableDeclined, 'xmark');
+    } else {
+      return pill(i18n.coteacherError(), styles.tableError, 'xmark');
+    }
+  };
 
-  const isAddDisabled = numCoteachers >= 5;
+  const tableRow = (index, coteacher) => {
+    return (
+      <tr key={index} className={styles.tableRow}>
+        <td className={styles.tableInfoCell}>
+          <div>
+            {coteacher.instructorName ? (
+              <>
+                <StrongText> {coteacher.instructorName}</StrongText>
+                <br />
+              </>
+            ) : null}
+
+            {coteacher.instructorEmail}
+          </div>
+        </td>
+        <td className={styles.tableStatusCell}>
+          {statusPill(coteacher.status)}
+        </td>
+        <td>
+          <button
+            type="button"
+            onClick={() => setCoteacherToRemove(coteacher)}
+            className={styles.tableRemoveButton}
+          >
+            <i className={classNames('fa-solid fa-trash', styles.trashIcon)} />
+          </button>
+        </td>
+      </tr>
+    );
+  };
+
+  const removeCoteacher = coteacher => e => {
+    e.preventDefault();
+    if (!coteacher.id) {
+      // remove from coteachersToAdd
+      setCoteachersToAdd(existing =>
+        existing.filter(teacher => teacher !== coteacher.instructorEmail)
+      );
+      setCoteacherToRemove({});
+      return;
+    }
+    $.ajax({
+      url: `/api/v1/section_instructors/${coteacher.id}`,
+      method: 'DELETE',
+    })
+      .done(() => {
+        addRemovedCoteacher(coteacher.id);
+        setCoteacherToRemove({});
+      })
+      .fail(() => setCoteacherToRemove({}));
+  };
+
+  const removePopup = coteacher => {
+    if (_.isEmpty(coteacher)) {
+      return null;
+    }
+    return (
```

Comment:
If we used `null` instead of `{}` to represent the absence of a coteacher to remove, we would 
a) make this method slightly shorter, and
b) not need to bring lodash into this component.

```suggestion
    return coteacher && (
```